examples/aa-simple-dapp/src/components/Content.tsx

```diff
@@ -0,0 +1,326 @@
+"use client";
+import { AlchemyNFTAbi, alchNftContractAddress } from "@/config/alch-nft";
+import { AlchemyTokenAbi, alchTokenContractAddress } from "@/config/alch-token";
+import { useWalletContext } from "@/context/wallet";
+import { PluginType } from "@/hooks/usePluginManager";
+import { getPluginTypeFromAddress } from "@/shared/util";
+import { MSCA, SessionKeyPlugin, type Plugin } from "@alchemy/aa-accounts";
+import { UserOperationCallData } from "@alchemy/aa-core";
+import { useCallback, useState } from "react";
+import { Address, Hash, encodeFunctionData } from "viem";
+
+type MintStatus =
+  | "Mint"
+  | "Requesting"
+  | "Minting"
+  | "Received"
+  | "Error Minting";
+
+type PluginUoStatus = "Requesting" | "In Progress..." | "Done" | "Error";
+
+export default function Content() {
+  const {
+    isLoggedIn,
+    provider,
+    scaAddress,
+    installedPlugins,
+    availablePlugins,
+    pluginInstall,
+    pluginUninstall,
+    refetchInstalledPlugins,
+    tokenReceiverEnabled,
+    sessionKey,
+    sessionKeySigner,
+    sessionKeyEnabled,
+  } = useWalletContext();
+  const [mintTxHash, setMintTxHash] = useState<Hash>();
+  const [mintStatus, setMintStatus] = useState<MintStatus>("Mint");
+  const [mintType, setMintType] = useState<"erc20" | "erc721" | null>(null);
+
+  const [pluginTxHash, setPluginTxHash] = useState<Hash>();
+  const [pluginTxStatus, setPluginTxStatus] = useState<PluginUoStatus | null>(
+    null
+  );
+  const [pluginType, setPluginType] = useState<PluginType | null>(null);
+
+  const handleMint = useCallback(
+    async (nft: boolean = false) => {
+      if (!provider) {
+        throw new Error("Provider not initialized");
+      }
+
+      if (nft && !tokenReceiverEnabled) {
+        throw new Error("Account requires TokenReceiverPlugin installed first");
+      }
+
+      try {
+        setMintTxHash(undefined);
+        setMintStatus("Requesting");
+        setMintType(nft ? "erc721" : "erc20");
+        const userOpData: UserOperationCallData = {
+          target: nft ? alchNftContractAddress : alchTokenContractAddress,
+          data: encodeFunctionData({
+            abi: nft ? AlchemyNFTAbi : AlchemyTokenAbi,
+            functionName: nft ? "mintTo" : "mint",
+            args: [await provider.getAddress()],
+          }),
+          value: BigInt(0),
+        };
+
+        const executeWithSessionKey =
+          provider.isConnected<MSCA>() && sessionKeyEnabled && sessionKey;
+        console.log("executing with session key", executeWithSessionKey);
+        const uoHash = executeWithSessionKey
+          ? await provider
+              .extend(SessionKeyPlugin.providerMethods)
+              .executeWithSessionKey(
+                {
+                  args: [
+                    [{ ...userOpData, value: userOpData.value || BigInt(0) }],
+                    sessionKey,
```

Comment:
isn't this the same as `sessionKeySigner.getAddress()`?

It feels weird that the `executeWithSessionKey` takes in just the session key address but not the session key to use as well? Ah but that's because of how it's autogenned

having to call `executeWithSessionKey` maybe isn't the cleanest devex IMO. Can we leverage the builder pattern to have a method like `createAccountWithSessionKey` or something that makes this even simpler? That could return a built out MSCA using `withExecutor` which handles fallback logic of either encoding `execute` or `executeWithSessionKey`.